torch/utils/benchmark/utils/timer.py

```diff
@@ -135,10 +189,25 @@ def __init__(
         description: Optional[str] = None,
         env: Optional[str] = None,
         num_threads: int = 1,
+        language: str = PYTHON,
     ):
         if not isinstance(stmt, str):
             raise ValueError("Currently only a `str` stmt is supported.")
 
+        self._language = language
+        if language not in (PYTHON, CPP):
+            raise ValueError(f"Invalid language `{language}`. (Choices: {PYTHON}, {CPP})")
+
+        if language == CPP:
+            assert self._timer_cls is timeit.Timer, "_timer_cls has already been swapped."
+            self._timer_cls = cast(
+                # CPPTimer does not inherit from timeit.Timer, however it implements
+                # `timeit` which is sufficient for our purposes.
```

Comment:
Do you want a Protocol here? https://mypy.readthedocs.io/en/stable/protocols.html

Reply:
Yes. That is **exactly** what I wanted.